server/src/main/java/org/elasticsearch/action/admin/indices/template/post/TransportSimulateIndexTemplateAction.java

```diff
@@ -217,7 +217,11 @@ public static Template resolveTemplate(
                 // the context is only used for validation so it's fine to pass fake values for the
                 // shard id and the current timestamp
                 tempIndexService.newSearchExecutionContext(0, 0, null, () -> 0L, null, emptyMap()),
-                tempIndexService.dateMathExpressionResolverAt()
+                tempIndexService.dateMathExpressionResolverAt(),
+                // we don't have a system indices object here, so we assume
+                // the alias is not a system name.
+                // TODO: any way to get system names here?
```

Comment:
Note to reviewer: does this need to be fixed? If so, can we do it in a follow-up?